### DISCUSSION TITLE
How to correctly set num_nodes for a PairData object?

### DISCUSSION THREAD

**Discussion by kyriakosschwarz** [pyg-team/pytorch_geometric#3297] [Q&A] [answered]
For a PairData object I get this warning:

`UserWarning: Unable to accurately infer 'num_nodes' from the attribute set '{'edge_attr_a', 'x_b', 'x_a', 'id', 'edge_index_a', 'edge_index_b', 'expression', 'edge_attr_b', 'y'}'. Please explicitly set 'num_nodes' as an attribute of 'data' to suppress this warning`

For batching purposes, how do I correctly set the 'num_nodes' attribute? Just the sum of nodes of both graphs, or something else?

Thanks a lot for any help!

**Comment by rusty1s** [comment, marked as answer]
Thanks for reporting. PyG raises this warning as it fails to compute the `batch` vector, which you do not need in your case anyways. As such, you can ignore the warning as it will not do any harm in your case.

Nonetheless, I fixed this issue in master, see https://github.com/pyg-team/pytorch_geometric/pull/3298. This should avoid printing the warning for `PairData` in the first place.

**Comment by kyriakosschwarz** [comment]
Great, thanks a lot!
